android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUITimePicker.java

```diff
@@ -160,15 +174,36 @@ public void setValue(long value, boolean suppressEvent)
 	@Override
 	public void onTimeChanged(TimePicker view, int hourOfDay, int minute)
 	{
+		detectTimeChange();
+	}
+	
+	private void detectTimeChange() {
 		Calendar calendar = Calendar.getInstance();
-		calendar.set(Calendar.HOUR_OF_DAY, hourOfDay);
-		calendar.set(Calendar.MINUTE, minute);
-		if (!suppressChangeEvent) {
+		TimePicker picker = (TimePicker) getNativeView();
+		boolean hasChanged;
+		if (Build.VERSION.SDK_INT >= 23) {
+			hasChanged = lastSelectedHour != picker.getHour() || lastSelectMin != picker.getMinute();
```

Comment:
@jquick-axway I agree if it were just a couple of instances, but if we plan on suppressing all of the warnings from our code base like you mentioned with [TIMOB-20258](https://jira.appcelerator.org/browse/TIMOB-20258) then it would be more maintainable to create a single class for each API level needed (`23, 21, 16`) in an appropriate place and reference to that; than having nested classes in each class with an unsupported API reference.

Here's a list of references we need to "hide" from Dalvik:
```
AndroidModule.java
API-23
 - L365: checkSelfPermission
 - L389: checkSelfPermission
 - L396: requestPermissions

CalendarProxy.java
API-23
 - L105: checkSelfPermission
 - L106: checkSelfPermission

CommonContactsApi.java
API-23
 - L72: checkSelfPermission

FilesystemModule.java
API-23
 - L94: checkSelfPermission
 - L110: requestPermissions

GeolocationModule.java
API-23
 - L623: checkSelfPermission
 - L645: requestPermissions

 MediaModule.java
API-23
  - L359: checkSelfPermission
  - L360: checkSelfPermission
  - L371: checkSelfPermission
  - L382: checkSelfPermission

TiUIWebView.java
- API-16
 - L331: setAllowUniversalAccessFromFileURLs

TiWebChromeClient.java
 API-23
  - L229: checkSelfPermission

TiWindowProxy.java
API-21
 - L531: makeSceneTransitionAnimation

WindowProxy.java
API-16
 - L162: startActivity
API-21
 - L176: finishAfterTransition
 - L493: setEnterTransition, createTransition
 - L497: setExitTransition, createTransition
 - L501: setReturnTransition, createTransition
 - L505: setReenterTransition, createTransition
 - L509: setSharedElementEnterTransition, createTransition
 - L513: setSharedElementExitTransition, createTransition
 - L517: setSharedElementReenterTransition, createTransition
 - L521: setSharedElementReturnTransition, createTransition

TiViewProxy.java
API-21
 - L802: createCircularReveal
 - L836: createCircularReveal

TiBaseActivity.java
API-23
 - L619: finishAndRemoveTask
 - L881: finishAffinity

TiActivitySupportHelper.java
API-16
 - startIntentSenderForResult

TiFileHelper2.java
API-23
 - L94: checkSelfPermission

TiBorderWrapperView.java
API-21
 - L128: ViewOutlineProvider

TiUIView.java
API-16
 - L545: removeOnGlobalLayoutListener
 - L628: removeOnGlobalLayoutListener
```
I don't like the idea of creating a nested class for each of those instances. We could do something like this:
```
org.appcelerator.titanium.compat
  Api23.java
  Api21.java
  Api16.java
```

These classes would be for internal use only, undocumented and not to be used by module developers.

However, I did notice even the Google support libraries do not suppress all of their warnings:
```
Could not find method android.content.res.Resources.getDrawable, referenced from method android.support.v7.widget.ResourcesWrapper.getDrawable
VFY: unable to resolve virtual method 565: Landroid/content/res/Resources;.getDrawable (ILandroid/content/res/Resources$Theme;)Landroid/graphics/drawable/Drawable;
VFY: replacing opcode 0x6e at 0x0002
Could not find method android.content.res.Resources.getDrawableForDensity, referenced from method android.support.v7.widget.ResourcesWrapper.getDrawableForDensity
VFY: unable to resolve virtual method 567: Landroid/content/res/Resources;.getDrawableForDensity (IILandroid/content/res/Resources$Theme;)Landroid/graphics/drawable/Drawable;
VFY: replacing opcode 0x6e at 0x0002
Could not find class 'android.support.graphics.drawable.VectorDrawableCompat', referenced from method android.support.v7.widget.AppCompatDrawableManager.isVectorDrawable
VFY: unable to resolve instanceof 365 (Landroid/support/graphics/drawable/VectorDrawableCompat;) in Landroid/support/v7/widget/AppCompatDrawableManager;
VFY: replacing opcode 0x20 at 0x0000
Could not find method android.support.graphics.drawable.VectorDrawableCompat.createFromXmlInner, referenced from method android.support.v7.widget.AppCompatDrawableManager$VdcInflateDelegate.createFromXmlInner
VFY: unable to resolve static method 1631: Landroid/support/graphics/drawable/VectorDrawableCompat;.createFromXmlInner (Landroid/content/res/Resources;Lorg/xmlpull/v1/XmlPullParser;Landroid/util/AttributeSet;Landroid/content/res/Resources$Theme;)Landroid/support/graphics/drawable/VectorDrawableCompat;
VFY: replacing opcode 0x71 at 0x0004
Could not find method android.support.graphics.drawable.AnimatedVectorDrawableCompat.createFromXmlInner, referenced from method android.support.v7.widget.AppCompatDrawableManager$AvdcInflateDelegate.createFromXmlInner
VFY: unable to resolve static method 1630: Landroid/support/graphics/drawable/AnimatedVectorDrawableCompat;.createFromXmlInner (Landroid/content/Context;Landroid/content/res/Resources;Lorg/xmlpull/v1/XmlPullParser;Landroid/util/AttributeSet;Landroid/content/res/Resources$Theme;)Landroid/support/graphics/drawable/AnimatedVectorDrawableCompat;
VFY: replacing opcode 0x71 at 0x0004
``` 

- What are your thoughts for this approach?
- Is it worthwhile to suppress these warnings?